private/model/api/shape.go

```diff
@@ -219,7 +224,7 @@ func (ref *ShapeRef) GoTypeWithPkgName() string {
 
 // Returns a string version of the Shape's type.
 // If withPkgName is true, the package name will be added as a prefix
-func goType(s *Shape, withPkgName bool) string {
+func goType(ref *ShapeRef, s *Shape, withPkgName bool) string {
```

Comment:
Instead of adding an optional ShapeRef. Is it possible to add this decoration somewhere else? Mainly because this is an optional parameter that is only used in a narrow use case.

Maybe compare this logic to how the SDK generates io.ReadSeeker, vs []byte payloads.